src/cobra/manipulation/delete.py

```diff
@@ -57,113 +61,43 @@ def prune_unused_reactions(model: "Model") -> Tuple["Model", List["Reaction"]]:
-        "It will be removed soon.",
-        DeprecationWarning,
-    )
-    return {r: r.gpr for r in model.reactions}
-
-
-def find_gene_knockout_reactions(
+def knock_out_model_genes(
     model: "Model",
-    gene_list: List["Gene"],
-    compiled_gene_reaction_rules: Optional[Dict["Reaction", Module]] = None,
+    gene_list: Union[List["Gene"], Set["Gene"], List[str], Set[str]],
 ) -> List["Reaction"]:
-    """Identify reactions which will be disabled when genes are knocked out.
+    """Disable the genes in `gene_list`.
+
+    It sets the bounds to "zero" for reactions catalysed by the genes in
+    `gene_list` if deleting the genes would stop the reactions from proceeding.
+
+    The changes are reverted upon exit if executed within the model as context.
 
     Parameters
     ----------
     model: cobra.Model
-        The model for which to find gene knock-out reactions.
+        The model whose reaction bounds are to be set.
     gene_list: list of cobra.Gene
         The list of genes to knock-out.
-    compiled_gene_reaction_rules: dict of {reaction: compiled_string},
-                                  optional
-        If provided, this gives pre-compiled gene-reaction rule strings.
-        The compiled rule strings can be evaluated much faster. If a rule
-        is not provided, the regular expression evaluation will be used.
-        Because not all gene-reaction rule strings can be evaluated, this
-        dict must exclude any rules which can not be used with eval
-        (default None).
 
     Returns
     -------
-    list of cobra.Reaction
-       The list of cobra.Reaction objects which will be disabled.
-
-    .. deprecated:: 0.22.1
-        Internal function that has outlived its purpose.
-
+    reaction_list: list[cobra.Reaction]
+        A list of cobra.Reactions that had the bounds turned to zero.
     """
-    warn(
-        "The function `find_gene_knockout_reactions` has outlived its purpose. "
-        "It will be removed in the next minor version (0.23.0).",
-        DeprecationWarning,
-    )
-    potential_reactions = set()
+    orig_bounds = model.reactions.list_attr("bounds")
     for gene in gene_list:
         if isinstance(gene, str):
             gene = model.genes.get_by_id(gene)
-        potential_reactions.update(gene._reaction)
-    gene_set = {str(i) for i in gene_list}
-    if compiled_gene_reaction_rules is None:
-        compiled_gene_reaction_rules = {r: r.gpr for r in potential_reactions}
-
-    return [
-        r
-        for r in potential_reactions
-        if not compiled_gene_reaction_rules[r].eval(gene_set)
-    ]
+        gene.knock_out()
+    new_bounds = model.reactions.list_attr("bounds")
+    reaction_list = list()
+    for i, rxn in enumerate(model.reactions):
+        if not (
+            np.isclose(orig_bounds[i][0], new_bounds[i][0])
+            and np.isclose(orig_bounds[i][1], new_bounds[i][1])
+        ):
+            reaction_list.append(rxn)
+    return reaction_list
```

Comment:
Even though it increases the computational work, I think a version more in line with the object-oriented design is:

```suggestion
    return [rxn for rxn in itertools.chain.from_iterable((gene.reactions for gene in gene_list)) if not rxn.functional]
```

Reply:
Why are there two parenthesis?

Reply:
Also, why set?

Reply:
Not sure gene.reactions as written would work, since it can accept strings

Reply:
Pushed a version with list comprehension that works. You can pick if the most recent one [
24cbb5e] makes sense , or [66d0428](https://github.com/opencobra/cobrapy/pull/1208/commits/66d04288a2e7abe6456ca64c73f19b104f5f0276) makes more sense.

Reply:
Double parenthesis was outer for the function call and the inner for a generator statement. This was assuming that `gene_list` would be the output of `get_by_any`, sorry, my comment wasn't clear.

I used a set to avoid duplicates which are in principle possible with a list and `get_by_any`.

Reply:
for gene in model.genes.get_by_any(set(gene_list)): fails due to type error

>       return the_id in self._dict
E       TypeError: unhashable type: 'set'

../manipulation/delete.py:86: in knock_out_model_genes
    for gene in model.genes.get_by_any(set(gene_list)):
../core/dictlist.py:108: in get_by_any
    return [get_item(item) for item in iterable]
../core/dictlist.py:108: in <listcomp>
    return [get_item(item) for item in iterable]
../core/dictlist.py:101: in get_item
../core/dictlist.py:379: TypeError

Reply:
I think the current version [https://github.com/opencobra/cobrapy/pull/1208/commits/ae1fdbe5beea58cc453253111914a1bd27e7b750] is the best, and object-oriented. Fixed other comments.